iModelJsNodeAddon/JsCloudSqlite.cpp

```diff
@@ -435,6 +435,14 @@ struct JsCloudContainer : CloudContainer, Napi::ObjectWrap<JsCloudContainer> {
         });
     }
 
+    Napi::Value GetWriteLockExpiryTime(NapiInfoCR) {
+        RequireConnected();
+        BeJsDocument lockedBy;
+        ReadWriteLock(lockedBy); 
+        // Returns empty string if writeLock is empty.
+        return Napi::String::New(Env(), lockedBy[JSON_NAME(expires)].asString());
+    }
+
     void ReadWriteLock(BeJsDocument& out) {
         Statement stmt;
         auto rc = stmt.Prepare(m_containerDb, "SELECT value FROM bcv_kv WHERE name=?");

```

Comment:
Is there a reason you cannot do this from the typescript side? It will save from adding new method to addon. If its just a select query you can do it from typescript.

Reply:
I don't believe I can do this from typescript because  bcv_kv is a virtual table, and we don't let virtual tables be queried freely. The connection in question must call sqlite3_bcvfs_register_vtab() ( from the c++ side?). The only connection which currently registers these vtables is only available in the addon.

Reply:
@khanaffan okay to resolve this comment?